cypress/e2e/gui-tests/engageSphere.cy.js

```diff
@@ -0,0 +1,430 @@
+      { fixture: 'mediumCustomers' }
+    ).as('getMediumCustomers')
+
+    cy.get('[data-testid="filter"]').select('Medium')
+
+    cy.wait('@getMediumCustomers')
+
+    cy.get('tbody tr').should('have.length', 5)
+  })
+
+  it('filters by Enterprise size', () => {
+    cy.intercept(
+      'GET',
+      `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=Enterprise`,
+      { fixture: 'enterpriseCustomers' }
+    ).as('getEnterpriseCustomers')
+
+    cy.get('[data-testid="filter"]').select('Enterprise')
+
+    cy.wait('@getEnterpriseCustomers')
+
+    cy.get('tbody tr').should('have.length', 1)
+  })
+
+  it('filters by Large Enterprise size', () => {
+    cy.intercept(
+      'GET',
+      `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=Large%20Enterprise`,
+      { fixture: 'largeEnterpriseCustomers' }
+    ).as('getLargeEnterpriseCustomers')
+
+    cy.get('[data-testid="filter"]').select('Large Enterprise')
+
+    cy.wait('@getLargeEnterpriseCustomers')
+
+    cy.get('tbody tr').should('have.length', 3)
+  })
+
+  it('filters by Very Large Enterprise size', () => {
+    cy.intercept(
+      'GET',
+      `${Cypress.env('API_URL')}/customers?page=1&limit=10&size=Very%20Large%20Enterprise`,
+      { fixture: 'veryLargeEnterpriseCustomers' }
+    ).as('getVeryLargeEnterpriseCustomers')
+
+    cy.get('[data-testid="filter"]').select('Very Large Enterprise')
+
+    cy.wait('@getVeryLargeEnterpriseCustomers')
+
+    cy.get('tbody tr').should('have.length', 9)
+  })
+
+  it('Correctly downloads a list of customers as a CSV file', () => {
+    cy.contains('button', 'Download CSV').click()
+
+    cy.readFile('cypress/downloads/customers.csv').should('exist').then((fileContent) => {
+      expect(fileContent).to.not.be.null
+    })
+  })
+
+  it('Shows the contact info of a specific customer', () => {
+    cy.get('tbody tr').should('exist')
+    cy.get('[data-testid="filter"]').select('Small')
+
+    cy.contains('td', 'Jacobs Co').click()
+
+    cy.get('h2').should('exist').and('have.text', 'Customer Details');
+    cy.contains('Company name: Jacobs Co').should('be.visible')
+  });
+
+  it('Shows "No contact info available" for a customer without contact information', () => {
+    cy.get('[data-testid="filter"]').select('Small')
+    cy.contains('td', 'Jacobs Co').click()
+
+    cy.contains('No contact info available').should('be.visible')
+  });
+
+  it('Shows customer address', () => {
+    cy.get('[data-testid="filter"]').select('Small')
+    cy.contains('td', 'Jacobs Co').click()
+
+    cy.get('.show-address-btn').click()
+    cy.contains('h3', 'Address').should('be.visible')
+  });
+
+  it('Hides customer address', () => {
+    cy.get('[data-testid="filter"]').select('Small')
+    cy.get('table tbody tr').should('be.visible');
+    cy.contains('td', 'Jacobs Co').click()
+
+    cy.get('.show-address-btn').click()
+    cy.get('h3').should('exist').and('have.text', 'Address')
```

Comment:
Essa linha poderia ser simplificada par ao seguinte:

```suggestion
    cy.contains('h3', 'Address').should('be.visible')
```